crates/ra_assists/src/fill_match_arms.rs

```diff
@@ -8,13 +8,39 @@ use ra_syntax::ast::{self, AstNode};
 
 use crate::{AssistCtx, Assist, AssistId};
 
+fn is_trivial_arm(arm: &ast::MatchArm) -> bool {
+    for (i, p) in arm.pats().enumerate() {
+        if i > 0 {
+            return false;
+        }
+
+        match p.kind() {
+            ast::PatKind::PlaceholderPat(_) => {}
+            _ => {
+                return false;
+            }
+        };
+    }
+    return true;
+}
+
 pub(crate) fn fill_match_arms(mut ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
     let match_expr = ctx.node_at_offset::<ast::MatchExpr>()?;
 
     // We already have some match arms, so we don't provide any assists.
+    // Unless if there is only one trivial match arm possibly created
+    // by match postfix complete. Trivial match arm is the catch all arm.
     match match_expr.match_arm_list() {
-        Some(arm_list) if arm_list.arms().count() > 0 => {
-            return None;
+        Some(arm_list) => {
```

Comment:
Thus is a bit unidiomatic. `.filter(is_non_trivial).skip(1).next().is_some()`, maybe?.

Reply:
Yeah, it's a pain to write this nicely without try blocks. I'd done something like this probably:

```
use itertools::Itertools::*;

fn is_trivial_arm(arm: &ast::MatchArm) -> bool {
    fn single_pattern(arm: &ast::MatchArm) -> Option<ast::PatKind> {
        let (pat, ) = arm.pats().collect_tuple()?;
        Some(pat.kind())
    }
    match single_pattern(arm) { Some(ast::PatKind::Placehoder::Pat(..)) => true, _ => false}
}
```

Reply:
I'm not sure how to do this with a chain of adapters. I don't think filter() would work (what if there are multiple catch all arms?). 

I tried this but I'm not sure how to differentiate between no arms and and more than one arm
```
        Some(arm_list) => {
            let (arm,) = arm_list.arms().collect_tuple()?;
            if !is_trivial_arm(arm) {
                return None;
            }
        }
```

Reply:
Ah, indeed this case is differenet, because we need to handle either zero or one arm, and not exactly a single arm. Sorry for leading you into the false direction :sweat_smile: